rag/vector_store.py

```
@@ -90,35 +91,69 @@ def _sanitize_collection_name(name: str) -> str:
     return cleaned[:63]
 
 
+def _company_hash(company_original: Optional[str]) -> str:
+    """サニタイズ前の企業名をNFKC正規化してSHA-256ハッシュの短縮値を返す (#938)。
+
+    `_sanitize_company_name_for_query` は許可文字以外を区切りなしで単純除去するため、
+    表記の異なる企業名（例: "H.I.S." と "HIS"）が同一のサニタイズ結果へ収束しうる。
```

Ruff (0.16.1)

[warning] Docstring contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF002)